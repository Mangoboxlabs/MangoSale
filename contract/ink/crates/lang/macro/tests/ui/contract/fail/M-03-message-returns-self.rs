use ink_lang as ink;

#[ink::contract]
mod message_returns_self {
    #[ink(storage)]
    pub struct MessageReturnsSelf {}

    impl MessageReturnsSelf {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn returns_self(&self) -> Self {}
    }
}

fn main() {}
