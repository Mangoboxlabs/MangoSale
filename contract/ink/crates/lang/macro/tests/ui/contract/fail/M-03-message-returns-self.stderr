error: ink! messages must not return `Self`
  --> $DIR/M-03-message-returns-self.rs:15:39
   |
15 |         pub fn returns_self(&self) -> Self {}
   |                                       ^^^^
